### PR TITLE
LPS-129306 Add autoComplete attribute to input-date, input-time and input-checkbox taglibs

### PR DESCRIPTION
**Pull request by antonio-ortega**
Hi guys,

Can you review this pull request, please?

As you can guess this is coming from the same customer than [LPS-128745](https://issues.liferay.com/browse/LPS-128745).

Once they've got that solution, they realised it was not working with date fields. Just by having a look at the code I saw it could happen as well with time and checkbox fields.

Because of that I'm adding this attribute to those three taglibs: input-date, input-time and input-checkbox.

Additionally, I'm updating the use we make of them within input-field to propagate autoComplete value.

Let me know if you have further questions.

Thanks.

Regards.

### CODE REVIEW

**Review by wincent — APPROVED** [review]
This all looks fine to me, but me reviewing JSP and Java changes is a bit like this:

dumb-and-dumber

If CI is green and @jbalsas doesn't raise any objections I'd forward this.

**Comment by jbalsas** [comment]
Looks good to me, but in the semantic version test there's a [compile JSP error](https://test-1-7.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/437475/AXIS_VARIABLE=0,label_exp=!master/console) in `input_date/page.jsp`:

> java.util.concurrent.ExecutionException: org.apache.jasper.JasperException: file:/opt/dev/projects/github/liferay-portal/portal-web/docroot/html/taglib/ui/input_date/page.jsp (line: [122], column: [14]) **Unterminated [&lt;aui:input] tag**

@antonio-ortega, can you double check if some `<` or `>` is unbalanced causing tags not to close properly?

File: portal-web/docroot/html/taglib/ui/input_date/page.jsp [hunk 1]
```diff
@@ -115,10 +116,10 @@ else {
...
+			<input <%= Validator.isNotNull(autoComplete) ? "autocomplete=\"" + autoComplete + "\"" : StringPool.BLANK %> class="form-control <%= cssClass %>" <%= disabled ? "disabled=\"disabled\"" : "" %> id="<%= nameId %>" name="<%= namespace + HtmlUtil.escapeAttribute(name) %>" type="date" value="<%= dateString %>" />
 		</c:when>
 		<c:otherwise>
-			<aui:input cssClass="<%= cssClass %>" disabled="<%= disabled %>" id="<%= HtmlUtil.getAUICompatibleId(name) %>" label="" name="<%= name %>" placeholder="<%= StringUtil.toLowerCase(placeholderValue) %>" required="<%= required %>" title="" type="text" value="<%= dateString %>" wrappedField="<%= true %>">
+			<aui:input <%= Validator.isNotNull(autoComplete) ? "autocomplete=\"" + autoComplete + "\"" : StringPool.BLANK %> cssClass="<%= cssClass %>" disabled="<%= disabled %>" id="<%= HtmlUtil.getAUICompatibleId(name) %>" label="" name="<%= name %>" placeholder="<%= StringUtil.toLowerCase(placeholderValue) %>" required="<%= required %>" title="" type="text" value="<%= dateString %>" wrappedField="<%= true %>">
```
Comment (jbalsas) [inline portal-web/docroot/html/taglib/ui/input_date/page.jsp:122]: It's probably this 😂

**Comment by antonio-ortega** [comment]
ci:test:sf

**Comment by antonio-ortega** [comment]
ci:test:relevant

**Comment by jbalsas** [comment]
ci:forward

**Pull request closed without merging**